contracts/src/LightClient.sol

```diff
@@ -341,4 +380,103 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice checks the L1BlockUpdates array and the delayThreshold to determine if hotshot was
+    /// down at a specified L1 block number
+    /// @param l1BlockNumber This L1 block number used to reference a point in time when this light
+    /// client contract was expected to be updated in a given threshold
+    /// @param delayThreshold The delay threshold used to determined if this Light Client contract
+    /// was updated in the desired time (measured in blocks)
+    function wasL1Updated(uint256 l1BlockNumber, uint256 delayThreshold)
+        public
+        view
+        returns (bool)
+    {
+        /**
+         * TODO: consider a revert if delayThreshold is invalid
+         */
+        uint256 updatesCount = l1BlockUpdates.length;
+
+        // Handling Edge Cases
+        // Edgecase 1: The block is in the future or in the past before HotShot was live
+        if (l1BlockNumber > block.number || (updatesCount > 0 && l1BlockNumber < l1BlockUpdates[0]))
+        {
+            revert InvalidL1BlockForCheckingL1Updates();
+        }
+
+        // Edgecase 2: There have only been two HotShot updates so far
+        // we only start checking if HotShot is live after it has received atleast two updates
+        if (updatesCount < 2) {
+            return true;
+        }
+
+        for (uint256 i = 0; i < updatesCount; i++) {
+            uint256 blockStart = l1BlockUpdates[i];
+            uint256 blockEnd = (i < updatesCount - 1) ? l1BlockUpdates[i + 1] : block.number;
```

Comment:
Maybe this code would be easier to read if the iteration only went until `i < updatesCount - 1` and the last case was handled after the loop?

Reply:
resolving this as the implementation has since changed